zerver/tests/test_subs.py

```diff
@@ -3514,6 +3514,7 @@ def test_compose_views_rollback(self) -> None:
         user_profile = self.example_user("hamlet")
         user_profile.full_name = "Hamlet"
         user_profile.save()
+        request = HttpRequest()
 
```

Comment:
I wasn't sure what to do in this case because what's being tested is the Django rollback, which should not make these changes. But I can imagine this change being an issue when `json_success` is updated to use the request data.

Reply:
I think we have a standard class, `HostRequestMock`, for this purpose.